src/tests/routes/controllers/social-auth.test.js

```diff
@@ -0,0 +1,121 @@
+} from '../../../routes/services/passport-strategies.services';
+
+chai.use(chaiHttp);
+
+describe('Social authentication', () => {
+  let facebookTestUser;
+  let twitterTestUser;
+  describe('Passport strategies setup', () => {
+    const accessToken = '',
+      refreshToken = '',
+      token = '',
+      tokenSecret = '',
+      done = (err, data = null) => {
+        if (data) {
+          return data;
+        }
+      };
+
+    const name = {
+      givenName: 'Uncle',
+      familyName: 'Musonant',
+    };
+    it('Should authenticate user profile from facebook', async () => {
+      const profile = {
+        id: 'demoFbSocialId',
+        name,
+        emails: [{ value: 'testMail.fb.com' }],
+        username: 'testFbUsername',
+        provider: 'facebook',
+      };
+      facebookTestUser = await facebookAuth(
+        accessToken,
+        refreshToken,
+        profile,
+        done,
+      );
+      expect(facebookTestUser.dataValues.id).to.be.a('string');
+      expect(facebookTestUser.dataValues.lastName).to.eql('Musonant');
+      expect(facebookTestUser.dataValues.email).to.eql('testMail.fb.com');
+      expect(facebookTestUser.dataValues.username).to.eql('testMail.fb.com');
+    });
+    it('Should authenticate user profile from twitter', async () => {
+      const profile = {
+        id: 'demoTwSocialId',
+        name,
+        emails: [{ value: 'testMail.tw.com' }],
+        username: 'testTwUsername',
+        provider: 'twitter',
+      };
+      twitterTestUser = await twitterAuth(token, tokenSecret, profile, done);
+      expect(twitterTestUser.dataValues.id).to.be.a('string');
+      expect(twitterTestUser.dataValues.email).to.eql('testMail.tw.com');
+      expect(twitterTestUser.dataValues.username).to.eql('testMail.tw.com');
+    });
+  });
+  describe('Social authentication controller', () => {
+    describe('should process user received from passport authentication', async () => {
+      let req = {
+        user: {
+          dataValues: {
+            firstName: 'Musonant',
+            username: 'mrmusonant@gmail.com',
+            email: 'mrmusonant@gmail.com',
+            socialProvider: 'twitter',
+            socialId: '892165240644829185',
+            id: '3505d530-a60f-4fc2-840a-ca9e864b3f33',
+            lastName: null,
+            password: null,
+            bio: null,
+            interests: null,
+            role: 'AUTHOR',
+            image: null,
+            isEmailVerified: false,
+            createdAt: '2019-04-24T13:02:34.828Z',
+            updatedAt: '2019-04-24T13:02:34.828Z',
+          },
+          _options: {
+            isNewRecord: true,
+          },
+        },
+      };
+      const res = {
+        status(code) {
+          res.statusCode = code;
+          return res;
+        },
+        json(data) {
+          return { body: data, status: res.statusCode };
+        },
+      };
+
+      it('and return the appropriate response for a new user', async () => {
```

Comment:
could you look into the sentence construction here to make it more readable?

Reply:
This statement continues from the describe block at the top of it. When running tests, it shows up like this:
Social controller should process user received from passport authentication
    -  and return the appropriate response for a new user

Reply:
Nice